scripts/tests/jdownloader.sh

```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+# TODO
```

Comment:
If you delete this file and `box test jdownloader` runs just fine after an install, you don't even need this. The default tests can go far, test files are only needed for exceptions to the defaults.

Reply:
I did not realize that `box test jdownloader` would do anything without this file. I will have to look into the default operations of `box test`.

However, I am thinking I would like to add a custom test. Specifically, I'm thinking something to detect if their installation is accessible via the third party website. Since the login information for that will be stored locally in plain text anyways. This would give sysadmins a quick test to see if any of the login information was entered incorrectly on install, if their login data has changed during install, or if the installation is unaccessible.

Reply:
Fair point that I should write docs for the testing functionality.

I would say that we should not rely on using tools from systems outside of our control, if that portal goes down we need a rewrite on our end. As long as the services are up, and they're accessible internally and via nginx, we're good I would say.

Reply:
I would rather access it directly as well. I'm not actually sure there is an alternative though. As far as I know JDownloader's web UI can **only** be used via their site. I thought I might have seen something contrary to this in passing the other day though... I'll have to take another look around.

Reply:
We can just check if the ports are open and being listened to, not necessarily make some `curl`/HTTP requests to them. We have a basetest function for this. I would say that's _good enough_

Reply:
Oh alright then. That does sound like a better solution to me. Where might I find this function?

Reply:
Check this https://github.com/swizzin/swizzin/blob/master/sources/functions/tests, let me know if the inline docs aren't good enough

Reply:
++ https://swizzin.ltd/dev/structure/#test

Reply:
Thinking the default tests aren't going to cut it in this case.
```
# box test jdownloader
INFO	Running default test for jdownloader.
	This test is likely to fail in case the item is not a 
	standard service+nginx app
...	Checking jdownloader service is active
Unit jdownloader.service could not be found.
WARN	jdownloader.service not active!
...	Checking if port jdownloader is open and listened to
WARN	Couldn't guess port
...	Checking if port jdownloader is reachable via curl
WARN	Couldn't guess port
WARN	nginx not installed, cannot verify if jdownloader is 
	reachable through it. Skipping.
ERROR	Errors were encountered
	Please consult the above and/or check the log (less -R +G 
	/root/logs/tests-2021.07.30-08:25:13.txt)
```

Reply:
BUT... 
I did find these today, and they suggest that it might be possible to access JDownloader without using the third party website.
[A help topic suggesting that NGINX might be an option.](https://board.jdownloader.org/showthread.php?t=76156)
[API Docs for reference](https://my.jdownloader.org/developers/)

Reply:
Yeah checking the web UI might be more trouble than it's worth honestly.

Reply:
You might not be wrong there. It's also entirely possible that the JDownloader logs would be telling us if it failed anyways.